### ISSUE TITLE
Feature Request - species NOTNULL and id formula = A for Record Summary

### ISSUE THREAD

**Issue by DerekSikes** [ArctosDB/arctos#6035]
for the Catalog Record Summary tool (the new one, not the one that is fired from after a normal cat record search) 

https://arctos.database.museum/record_summary.cfm?guid_prefix=KNWR:Ento,KNWR:Inv,KWP:Ento,UAA:Ento,UAM:Ento,UAMObs:Ento&phylum=Arthropoda&state_prov=Alaska&result_columns=phylum,phylclass,phylorder,suborder,superfamily,family,subfamily,tribe,subtribe,genus,species

I'd like to restrict the search to

species = NOTNULL

and

ID formula = A

can those options get added to that tool?

oh, also I would like the taxonomy field Superorder

Thanks!

**Priority**
Please assign a priority-label. Unprioritized issues gets sent into a black hole of despair.
high

**Comment by dustymc** [comment]
Species and taxa formula are in FLAT so this isn't technically problematic, but species doesn't mean what most users will think and https://github.com/ArctosDB/arctos/issues/3540 is going to mean that the identification (and therefore formula) chosen for FLAT is essentially random. Neither of these seem like great candidates for public exposure - does anyone care?

**Comment by DerekSikes** [comment]
I care (but I posted the original request) - use case: the dynamic checklist of the nonmarine arthropods of Alaska (one of the few states with such a checklist) is generated from this. We have two forms of the checklist - a smaller list with only identifications of species = not NULL (= all specimens identified to species level) and a larger list that includes IDs to genus only etc. Also, the taxon formula is needed since =A is the smaller list, and = anything is for the larger list (which includes things identified as near certain species, or new species, etc.).

And we need Superorder because of the acarologists having split mites into 2 groups.

**Comment by dustymc** [comment]
https://arctos.database.museum/record_summary.cfm?genus=Rattus&species=_&taxa_formula==A&result_columns=genus,species

I can't add anything that's not in the cache, superorder will need its own issue (or https://github.com/ArctosDB/arctos/issues/5460, but I think that's come completely off the rails and is going nowhere).

**Comment by DerekSikes** [comment]
So the _ is how we specify notnull for species? thanks

**Comment by dustymc** [comment]
Yep, _ = "match any character" - and it should work anywhere(ish, probably) in that form.

**Comment by DerekSikes** [comment]
There's something wrong with this. Here's my shareable URL for our species-only list:

https://arctos.database.museum/record_summary.cfm?guid_prefix=KNWR:Ento,KNWR:Inv,KWP:Ento,UAA:Ento,UAM:Ento,UAMObs:Ento&phylum=Arthropoda&species=_&taxa_formula==A&state_prov=Alaska&result_columns=phylum,phylclass,phylorder,suborder,superfamily,family,subfamily,tribe,subtribe,genus,species,taxa_formula

I was first struck by how the list is now ~13k species whereas when I last checked it was about 9k species! I downloaded the list and notice that some species are listed twice, eg Larinioides patagiatus

This is not how it should be working. I want it to work as it did before - when summarizing this way it should only list a species once. Can you figure out what's wrong and fix?

**Comment by DerekSikes** [comment]
further investigation yields this info: that species I noticed as being listed twice has different higher taxonomy (different classifications) for the same species name:

18	Arthropoda Arachnida Araneae Opisthothelae Araneoidea Araneidae Larinioides Larinioides patagiatus	

31	Arthropoda Arachnida Araneae Araneoidea Araneidae Larinioides Larinioides patagiatus

How did that happen & how do I fix? And why did it not happen before *(and can we go back to those better times?) - maybe this summary form needs to know my preferred classification and use only that??

**Comment by DerekSikes** [comment]
for the record, 7812 of these species names occur more than once in that list. I want them to occur only once.

**Comment by dustymc** [comment]
>How did that happen & how do I fix?

Assuming it's a glitch in the matrix, I asked for a refresh.

There are a bunch of collections here - are you sure they all have the same preferences?

>summary form needs to know my preferred classification and use only that?

It does - that's how FLAT works. 

>7812 of these species names 

I'll ask for a bigger refresh

**Comment by DerekSikes** [comment]
I checked all those collections preferred taxonomies and most had only 1 (Arctos) but 2 of the collections (UAM:Ento and UAMObs:Ento) had 2 classifications listed (Arctos first, WoRMS via Arctos 2nd). I don't know if this was the cause but I've now changed those to have only 1 classification listed.

**Comment by DerekSikes** [comment]
Why is this closed? I am still seeing this problem. Will it be gone by tomorrow?

**Comment by dustymc** [comment]
https://arctos.database.museum/info/flat_status.cfm

**Comment by DerekSikes** [comment]
https://arctos.database.museum/record_summary.cfm?guid_prefix=KNWR:Ento,KNWR:Inv,KWP:Ento,UAA:Ento,UAM:Ento,UAMObs:Ento&phylum=Arthropoda&state_prov=Alaska&result_columns=phylum,phylclass,phylorder,suborder,superfamily,family,subfamily,tribe,subtribe,genus,species

is still returning duplicate species (albeit only 106 instead of 7k)...

**Comment by dustymc** [comment]
jiggled the wires again, I'm not sure why I'd have missed some on the previous update request but it seems like that's what happened, at least the weird stuff (null suborder is what I checked) haven't been updated for a year or so

**Comment by DerekSikes** [comment]
https://arctos.database.museum/record_summary.cfm?guid_prefix=KNWR:Ento,KNWR:Inv,KWP:Ento,UAA:Ento,UAM:Ento,UAMObs:Ento&phylum=Arthropoda&state_prov=Alaska&result_columns=phylum,phylclass,phylorder,suborder,superfamily,family,subfamily,tribe,subtribe,genus,species

is still returning duplicate species (albeit only 34 instead of 7k)...

**Comment by dustymc** [comment]
https://arctos.database.museum/info/flat_status.cfm

**Comment by DerekSikes** [comment]
it's better - that query returns 3 pairs of duplicates. I was checking one of the duplicate names which is an ID for 22 records and noticed this odd pair:
https://arctos.database.museum/guid/UAMObs:Ento:84972
https://arctos.database.museum/guid/UAMObs:Ento:84973

which appear to be identical in every way (except GUID) and one has an 'Identifiers and Relationships' panel and the other doesn't. What's up with that? [btw - I have no idea why I made such duplicate records & am now considering deleting some).

**Comment by DerekSikes** [comment]
Now this link is returning 13 pairs of duplicates (the other day it was 6 pairs).

https://arctos.database.museum/record_summary.cfm?guid_prefix=KNWR:Ento,KNWR:Inv,KWP:Ento,UAA:Ento,UAM:Ento,UAMObs:Ento&phylum=Arthropoda&state_prov=Alaska&result_columns=phylum,phylclass,phylorder,suborder,superfamily,family,subfamily,tribe,subtribe,genus,species

One of the duplicate pairs involves the name Polyphemus pediculus which in the 'Genus' field has 1 of the pair with the expected value 'Polyphemus' and the other with this: 'Scapholeberis; Polyphemus'

But when I go to taxonomy and search on Polyphemus pediculus I see all the classifications it is in & then search the page for Scapholeberis, it is not found! What is going on??

This duplicates problem is new - not sure when it started, but sometime before the roll out of Arctos 2.0 (or whatever we are calling this new version) it did not happen. The cause needs discovery and fixing.

**Comment by dustymc** [comment]
Try this:

https://arctos.database.museum/record_summary.cfm?guid_prefix=KNWR:Ento,KNWR:Inv,KWP:Ento,UAA:Ento,UAM:Ento,UAMObs:Ento&phylum=Arthropoda&species=_&taxa_formula==A&state_prov=Alaska&result_columns=phylum,phylclass,phylorder,suborder,superfamily,family,subfamily,tribe,subtribe,genus,species

**Comment by DerekSikes** [comment]
Ok, that works. I think & hope this is good!
